### PR TITLE
Added ArrayUtils 

### PR DESCRIPTION
**Pull request by rameeznaufal**

### CODE REVIEW

File: entry/src/main/ets/MainAbility/pages/ArrayUtilsSample.ets [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ArrayUtils } from '@ohos/utilcode';
+import prompt from '@system.prompt';
+import dataStorage from '@ohos.data.storage';
+import ability_featureAbility from '@ohos.ability.featureAbility';
+import dataRdb from '@ohos.data.rdb';
+
```
Comment (SunandhiniMuralidharan) [inline entry/src/main/ets/MainAbility/pages/ArrayUtilsSample.ets:21]: Remove unused imports

File: entry/src/main/ets/MainAbility/pages/ArrayUtilsSample.ets [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+    }
+  }
+
+  searchElementArray1( element ) {
+    let res = ArrayUtils.contains(this.array1, element);
+    this.data = element + " present in array1: " + res;
+    return;
+  }
+
+
```
Comment (SunandhiniMuralidharan) [inline entry/src/main/ets/MainAbility/pages/ArrayUtilsSample.ets:74]: Remove unnecessary empty lines in all places

File: entry/src/main/ets/MainAbility/pages/ArrayUtilsSample.ets [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+      Column() {
+        Row() {
+          Flex({ direction: FlexDirection.Column, alignItems: ItemAlign.Center, justifyContent: FlexAlign.SpaceBetween }) {
+            Flex({
+              direction: FlexDirection.Row,
+              alignItems: ItemAlign.Center,
+              justifyContent: FlexAlign.SpaceEvenly
+            }) {
+              TextInput({ placeholder: 'Array 1  (String values)' })
```
Comment (SunandhiniMuralidharan) [inline entry/src/main/ets/MainAbility/pages/ArrayUtilsSample.ets:102]: Format the code

File: entry/src/main/ets/MainAbility/pages/ArrayUtilsSample.ets [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+                .borderRadius(10)
+            }
+
+            Flex({
+              direction: FlexDirection.Row,
+              alignItems: ItemAlign.Center,
+              justifyContent: FlexAlign.SpaceEvenly
+            }) {
+
+              TextInput({ placeholder: 'Array 2 (String values)' })
```
Comment (SunandhiniMuralidharan) [inline entry/src/main/ets/MainAbility/pages/ArrayUtilsSample.ets:127]: Remove unnecessary empty lines - check for in all places

File: entry/src/main/ets/MainAbility/pages/ArrayUtilsSample.ets [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+                  })
+                  .margin(5)
+                  .borderRadius(10)
+
+                }
+            }.scrollable(ScrollDirection.Horizontal).scrollBar(BarState.On)
+             .scrollBarColor(Color.Gray).scrollBarWidth(100)
+
+              Text(this.data)
+                .fontSize(20)
```
Comment (SunandhiniMuralidharan) [inline entry/src/main/ets/MainAbility/pages/ArrayUtilsSample.ets:229]: Check formatting in all places

File: entry/src/main/ets/MainAbility/pages/ArrayUtilsSample.ets [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+  getLength() {
+    return 'array1: ' + ArrayUtils.getLength(this.array1) + ' | array2: ' + ArrayUtils.getLength(this.array2);
+  }
+
+  isEmpty() {
+    return 'array1: ' + ArrayUtils.isEmpty(this.array1) + ' | array2: ' + ArrayUtils.isEmpty(this.array2);
+  }
+
+  isLengthEq() {
+    return 'array1.len == array2.len : ' + ArrayUtils.isSameLength(this.array1, this.array2);
```
Comment (SunandhiniMuralidharan) [inline entry/src/main/ets/MainAbility/pages/ArrayUtilsSample.ets:46]: give full names to function

File: entry/src/main/ets/MainAbility/pages/ArrayUtilsSample.ets [hunk 7]
```diff
@@ -0,0 +1,283 @@
...
+  }
+
+  isEq() {
+    return ArrayUtils.equals(this.array1, this.array2) ? 'Both are equal' : 'Both are not equal';
+  }
```
Comment (SunandhiniMuralidharan) [inline entry/src/main/ets/MainAbility/pages/ArrayUtilsSample.ets:51]: give full names to functions

File: utilcode/src/main/ets/utils/ArrayUtils.ets [hunk 8]
```diff
@@ -0,0 +1,295 @@
...
+  static isEmpty(array) {
+    if (array)
+    return array.length == 0;
+    else
+    return true;
+  }
+
+  static getLength(array) {
+    if (array)
+    return array.length;
```
Comment (archanaatr) [inline utilcode/src/main/ets/utils/ArrayUtils.ets:35]: format all the ets files, open each file and perform ctrl + Alt + L to format the code.

File: utilcode/src/main/ets/utils/ArrayUtils.ets [hunk 9]
```diff
@@ -0,0 +1,295 @@
...
+    }
+    for (let i = 0; i < newSize; ++i) {
+     subArray = ArrayUtils.realAddOne<T>(subArray, array[startIndexInclusive + i]);
+    }
+    return subArray;
+  }
+
+  static add<T>(array: T[], element: T) {
+    if (array) {
+      return <T[]> ArrayUtils.realAddOne<T>(array, element);
```
Comment (archanaatr) [inline utilcode/src/main/ets/utils/ArrayUtils.ets:139]: recheck this logic, try to combine to one single condition.

File: utilcode/src/main/ets/utils/ArrayUtils.ets [hunk 10]
```diff
@@ -0,0 +1,280 @@
...
+  }
+
+  static readonly INDEX_NOT_FOUND = -1;
+
+  static newArray<Type>(array: Type[]): Type[] {
+    return <Type[]> array;
+  }
+
+  static isEmpty(array) {
+    if (array)
```
Comment (archanaatr) [inline utilcode/src/main/ets/utils/ArrayUtils.ets:27]: if (array) {
      return array.length == 0;
    } else {
      return true;
    }

include braces even for single line conditions.

File: utilcode/src/main/ets/utils/ArrayUtils.ets [hunk 11]
```diff
@@ -0,0 +1,280 @@
...
+  static getValue(array, index: number, defaultValue: number = null) {
+    if (array == null || array == undefined) return defaultValue;
+    try {
+      return array[index];
+    } catch (e) {
+      return defaultValue;
+    }
+  }
+
+  static setValue(array, index: number, value) {
```
Comment (archanaatr) [inline utilcode/src/main/ets/utils/ArrayUtils.ets:53]: Better to have validation for length of array also compared to index.

Recheck for this validation everywhere

File: utilcode/src/main/ets/utils/ArrayUtils.ets [hunk 12]
```diff
@@ -0,0 +1,280 @@
...
+    if (array == null || array == undefined) {
+      return null;
+    }
+
+    if (startIndexInclusive < 0) {
+      startIndexInclusive = 0;
+    }
+
+    let length = ArrayUtils.getLength(array);
+    if (endIndexExclusive > length) {
```
Comment (archanaatr) [inline utilcode/src/main/ets/utils/ArrayUtils.ets:123]: can have condition of endIndexExclusive < 0  included here

**Pull request merged**